department-of-veterans-affairs/caseflow
# Adds the Remand Received action to MDR & MandateHold Tasks

Bumps [CASEFLOW-98](https://vajira.max.gov/browse/CASEFLOW-98)

### Description
Adds the new Task Action `mandate received` and makes it available on the MDR task.

Stack PR for [CASEFLOW-98](https://vajira.max.gov/browse/CASEFLOW-98)

Part 1 - Adds the Remand Received action to MDR Task
Part 2 - TK
Part 3 - TK

### Acceptance Criteria
 - [ ] Work is behind the feature toggle: mdr_cavc_remand
 - [ ] Accessible to the CAVC Litigation Support team
 - [ ] MDR task has the `Remand Ready for 90 Day Letter` action
 - [ ] MandateHold task has the `Ready for VLJ` action
 - [ ] Both of these actions show the Add Mandate modal  

### Testing Plan
1. Create an MDR cavc remand
    `FactoryBot.create(:cavc_remand, :mdr).remand_appeal.uuid`
2. Sign in as a CAVC Lit Support user and nagivate to the appeal
   - [x] Confirm the action `Remand ready for 90-day letter`
   - [x] Confirm selection the action shows the entry modal
3. that's it for PR 1! submission is PR 2
4. Repeat for No mandate Straight Reversal
    `FactoryBot.create(:cavc_remand, :straight_reversal, :no_mandate).remand_appeal.uuid`
   - [ ] Confirm the action `Ready for VLJ`
   - [ ] Confirm selection the action shows the entry modal
5. Repeat for No mandate Death Dismissal
   `FactoryBot.create(:cavc_remand, :death_dismissal, :no_mandate).remand_appeal.uuid`
   - [ ] Confirm the action `Ready for VLJ`
   - [ ] Confirm selection the action shows the entry modal

### User Facing Changes

 BEFORE|AFTER
 ---|---
MDR | ![MDR case details page showing new action](https://user-images.githubusercontent.com/6129479/106944059-27c1d580-66f4-11eb-94d6-478d9b7f4768.png)
Straight Reversal & Death Dismissal | ![Death Dismissal case details page showing new action](https://user-images.githubusercontent.com/6129479/106945287-9e130780-66f5-11eb-91dd-40c7ff73d8c5.png)
